docs/approach-docs/platform-containers.md

```diff
@@ -0,0 +1,98 @@
+# A Containerized Approach Using Docker and Compose - Approach Doc
+
+I would like to propose the adoption of docker and docker-compose to begin containerizing the Journaly stack, starting with the database, paving the way for other servers or services that require installing software on a local development machine.
+
+## Details
+
+Installing postgres directly on a development machine might not be possible for some, and requires a level of support and maintenance on their part. With a containerized approach, we can eliminate this hurdle and perhaps shorten the time it takes for new developers to get setup. 
+
+Additionally, we could have a more consistent database environment for all developers, for instance, setting up users and other related artifacts during the containerization process.
+
+When I first setup Journaly, I used docker for my DB, and with the following changes, had it up and running in little time.
+
+### Docker
+
+We need a root level `Dockerfile` that looks something like this:
+
+```docker
+FROM postgres:11
+
+EXPOSE 5432
+
+COPY bootstrap.sh /docker-entrypoint-initdb.d/db-bootstrap.sh
+```
+
+You'll see that I have a bootstrap shell script to set up some initial database work:
+
+```sh
+#!/bin/bash
+
+set -e # bail immediately on error
+
+psql <<- EOSQL
```

Comment:
I like the approach of seeding the DB on build. `npm run db:snapshot` can be used to dump a development DB, we could check in the `snapshot.sql` file that generates and use that to seed the DB

Reply:
Great idea!